AttributeError with recent versions of Dask

With recent versions of Dask, trying to use Xpublish fails due to an `AttributeError` accessing `dask.array.Array` when only `import dask` is at the top level of the file.

```py
xpublish/utils/zarr.py:20: in <module>
    dask_array_type = (dask.array.Array,)
E   AttributeError: module 'dask' has no attribute 'array'
```

[It looks like `dask.array`](https://github.com/dask/dask/issues/8329) may need to be explicitly imported to access any Dask array functions.

I was able to reproduce by creating a fresh conda environment with python 3.9, and then pip installing `requirements.txt` and `dev-requirements.txt`. Then running `pytest` reproduced the error.

The issue is also shown in recent [Actions runs](https://github.com/xarray-contrib/xpublish/actions/runs/3370429909/jobs/5591319435#step:6:67).

<summary>Environment<summary>
<details>

```
➜ python --version
Python 3.9.13

➜ python -m pip list
Package            Version
------------------ ---------
anyio              3.6.2
appdirs            1.4.4
asciitree          0.3.3
attrs              22.1.0
cachey             0.2.1
certifi            2022.9.24
cftime             1.6.2
charset-normalizer 2.1.1
click              8.1.3
cloudpickle        2.2.0
coverage           6.5.0
dask               2022.11.0
entrypoints        0.4
exceptiongroup     1.0.4
fastapi            0.87.0
fasteners          0.18
fsspec             2022.11.0
h11                0.14.0
HeapDict           1.0.1
idna               3.4
iniconfig          1.1.1
locket             1.0.0
netCDF4            1.6.2
numcodecs          0.10.2
numpy              1.23.4
packaging          21.3
pandas             1.5.1
partd              1.3.0
pip                22.3.1
pluggy             1.0.0
pooch              1.6.0
pydantic           1.10.2
pyparsing          3.0.9
pytest             7.2.0
pytest-cov         4.0.0
pytest-mock        3.10.0
pytest-sugar       0.9.6
python-dateutil    2.8.2
pytz               2022.6
PyYAML             6.0
requests           2.28.1
setuptools         65.5.1
six                1.16.0
sniffio            1.3.0
starlette          0.21.0
termcolor          2.1.0
tomli              2.0.1
toolz              0.12.0
typing_extensions  4.4.0
urllib3            1.26.12
uvicorn            0.19.0
wheel              0.38.4
xarray             2022.11.0
zarr               2.13.3
```

</details>

--- a/dev-requirements.txt
+++ b/dev-requirements.txt
@@ -1,4 +1,5 @@
 fsspec
+httpx
 netcdf4
 pooch
 pytest

--- a/xpublish/utils/zarr.py
+++ b/xpublish/utils/zarr.py
@@ -1,7 +1,7 @@
 import copy
 import logging
 
-import dask
+import dask.array
 import numpy as np
 import xarray as xr
 from numcodecs.compat import ensure_ndarray

